### PR TITLE
Added transform "RandomCropFromBorders" and support for BBox in ElasticTransform, GridDistortion, OpticalDistortion

### PR DESCRIPTION
**Pull request by ZFTurbo**

### CODE REVIEW

**Comment by Dipet** [comment]
Thanks for your PR. Please update the README as written here https://github.com/albumentations-team/albumentations#contributing

**Comment by Dipet** [comment]
Also update code using black `black --config=black.toml`

**Comment by vladserkoff** [comment]
@ZFTurbo, could you also implement a `mask_from_bbox` as a function as well, as you've done with the `bbox_for_mask`. It would be really useful for other mask-only transforms.
P.S. I thinks it's better to rename `bbox_for_mask` to `bbox_from_mask`, makes more sense to me this way.

File: albumentations/augmentations/transforms.py [hunk 1]
```diff
@@ -794,6 +795,89 @@ def get_transform_init_args_names(self):
...
 
+class RandomCropFromBorders(DualTransform):
+    """Crop bbox from image randomly cut parts from borders without resize at the end
+
+    Args:
+        crop_value (float): float value in (0.0, 0.5) range. Default 0.1
+        crop_left (float): float value in (0.0, 1.0) range. Default 0.1
+        crop_right (float): float value in (0.0, 1.0) range. Default 0.1
+        crop_top (float): float value in (0.0, 1.0) range. Default 0.1
+        crop_bottom (float): float value in (0.0, 1.0) range. Default 0.1
```
Comment (Dipet) [inline albumentations/augmentations/transforms.py]: Please, add description that values randomly chooses from ranges `[0, crop_left * width)`, `[width - crop_right * width, width)` and etc.

**Comment by ternaus** [comment]
@ZFTurbo could you please rebase on the current master?

**Comment by IlyaOvodov** [comment]
People! Somebody remove blank line at end of file functional.py to fix tests and merge it! Very usefull PR!

**Pull request merged**